### PR TITLE
TkDQM: Add standalone monitoring modules for OT Rechits and new resolution plots for ITRechits

### PR DESCRIPTION
**Pull request by sroychow**
#### PR description:
This PR adds 

- Modules for OT Rechit monitoring and validation. These can be run only with a standalone config - DQM/SiTrackerPhase2/test/dqmstep_phase2tk_cfg.py. The input to the config should be a GEN-SIM-RECO file. It runs the IT and OT rechit producers together with the DQM & Validation step.
- Fixes an axis range in OT cluster validation module.
- For IT rechits, plots of  cluster res vs cluster size(in X & Y) are added. 
- For IT rechits, a scatter plot of res Y vs res X is added 

#### PR validation:
Validated with wf 23234.0, 34634.0.

cc @gabriel-rmrz @mmusich @tsusa @emiglior

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Validation/SiTrackerPhase2V/plugins/Phase2OTValidateRecHit.cc [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+        nrechitLayerMapS_primary[key] += DSViter->size();
+      }
+    }
+    edmNew::DetSet<Phase2TrackerRecHit1D>::const_iterator rechitIt;
+    //loop over rechits for a single detId
+    for (rechitIt = DSViter->begin(); rechitIt != DSViter->end(); ++rechitIt) {
+      LocalPoint lp = rechitIt->localPosition();
+      //GetSimHits
+      const std::vector<SimHitIdpr>& matchedId = associateRecHit.associateHitId(*rechitIt);
+      //std::cout << "Nmatched SimHits = "  << matchedId.size() << ",";
```
Comment (mmusich) [inline Validation/SiTrackerPhase2V/plugins/Phase2OTValidateRecHit.cc:231]: please remove or `LogDebug`

**Review by mmusich — COMMENTED** [review]
@sroychow thanks a lot for this!
I left few mostly technical comments below about modernized event and ES data access.

**Comment by sroychow** [comment]
@mmusich thanks for your comments. Addressed the comments and updated the other modules to the recommended syntax of getting Handle and ES data.

**Comment by mmusich** [comment]
> Addressed the comments and updated the other modules to the recommended syntax of getting Handle and ES data.

thank you!

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc [hunk 2]
```diff
@@ -99,20 +99,23 @@ void Phase2TrackerValidateDigi::dqmBeginRun(const edm::Run& iRun, const edm::Eve
...
 //
 void Phase2TrackerValidateDigi::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  // Get digis
-  iEvent.getByToken(itPixelDigiToken_, itPixelDigiHandle_);
-  iEvent.getByToken(otDigiToken_, otDigiHandle_);
+  //Digis
+  const auto& itPixelDigiHandle_ = iEvent.getHandle(itPixelDigiToken_);
+  const auto& otDigiHandle_ = iEvent.getHandle(otDigiToken_);
+  itdigis_ = itPixelDigiHandle_.product();
+  otdigis_ = otDigiHandle_.product();
```
Comment (mmusich) [inline Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc:112]:
```suggestion
  itdigis_ =  iEvent.get(itPixelDigiToken_);
  otdigis_ =  iEvent.get(otDigiToken_);
```

File: Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc [hunk 3]
```diff
@@ -99,20 +99,23 @@ void Phase2TrackerValidateDigi::dqmBeginRun(const edm::Run& iRun, const edm::Eve
...
 
   // DigiSimLink
-  iEvent.getByToken(itPixelDigiSimLinkToken_, itPixelSimLinkHandle_);
-  iEvent.getByToken(otDigiSimLinkToken_, otSimLinkHandle_);
+  const auto& itPixelSimLinkHandle_ = iEvent.getHandle(itPixelDigiSimLinkToken_);
+  const auto& otSimLinkHandle_ = iEvent.getHandle(otDigiSimLinkToken_);
+  otSimLink_ = otSimLinkHandle_.product();
+  itSimLink_ = itPixelSimLinkHandle_.product();
```
Comment (mmusich) [inline Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc:118]:
```suggestion
  otSimLink_ = iEvent.get(itPixelDigiSimLinkToken_);
  itSimLink_ = iEvent.get(otDigiSimLinkToken_);
```
would this work?

**Review by mmusich — COMMENTED** [review]
Further simplification according to: https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#get
Nitpicking here.

File: Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc (hunk 3 above)
Comment (sroychow) [reply Validation/SiTrackerPhase2V/plugins/Phase2TrackerValidateDigi.cc]: Please have a look at the update.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@sroychow there's still a couple of clang-tidy warnings: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfd2d0/14391/llvm-analysis/runCMSClangTidyChecks.log. 
Also since you are at it can you squash the last 8 commits?

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by davidlange6** [comment]
hi @sroychow @cms-sw/dqm-l2 - it looks like this PR adds a dependency on Validation to DQM/SiTrackerPhase2. Could that be fixed? Possibly by moving stuff that needs MC information to Validation. Thanks!

**Comment by sroychow** [comment]
@davidlange6 Thanks! should be addressed by #33557. Please have a look.

**Pull request merged**